File: 2645-pass-the-pillow/pass-the-pillow.c
int passThePillow(int n, int time) {
    int index = 1;
    int direction = 1;
    for(;time > 0; time--){
        index += direction;

        if(index == n || index == 1){
            direction = direction * -1;
        }
    }
    return index;
}

File: 2645-pass-the-pillow/pass-the-pillow.js
/**
 * @param {number} n
 * @param {number} time
 * @return {number}
 */
var passThePillow = function(n, time) {
    let index = 1;
    while(time > 0){
        while(index < n && time > 0){
            index++;
            time--;
        }
        while(index > 1 && time > 0){
            index--;
            time--;
        }
    }
    return index;
};